File: frontend/src/types/media.ts
export type MediaType = "Image" | "Video";
export type MediaCategory = "Advertisement" | "Promotion" | "Branding" | "Announcement" | "Emergency";

export const MEDIA_CATEGORIES: MediaCategory[] = ["Advertisement", "Promotion", "Branding", "Announcement", "Emergency"];

export interface MediaItem {
  id: string;
  name: string;
  type: MediaType;
  category: MediaCategory;
  thumbnail: string;
  originalFile: string;
  size: number;
  dimensions: string;
  duration?: number | null;
  uploadedAt: number;
  uploadedBy: string;
  checksum?: string | null;
  /** Owning client; null or absent means it belongs to the operator. Only administrators can change it. */
  clientId?: string | null;
}

export interface MediaUpdatePayload {
  name?: string;
  category?: MediaCategory;
  duration?: number;
  dimensions?: string;
  clientId?: string | null;
}

File: backend/app/models/media.py
from sqlalchemy import Column, String, BigInteger, Integer
from app.database.base import Base

class Media(Base):
    __tablename__ = "media"

    id = Column(String, primary_key=True, index=True)
    name = Column(String, nullable=False)
    type = Column(String, nullable=False)
    category = Column(String, nullable=False)
    thumbnail = Column(String, nullable=False)
    originalFile = Column(String, nullable=False)
    size = Column(BigInteger, nullable=False)
    dimensions = Column(String, nullable=False)
    duration = Column(Integer, nullable=True)
    uploadedAt = Column(BigInteger, nullable=False)
    uploadedBy = Column(String, nullable=False)
    checksum = Column(String, nullable=True)